src/middlewared/middlewared/plugins/smb_/status.py

```diff
@@ -54,33 +54,11 @@ def status(self, info_level, filters, options, status_options):
         """
         lvl = InfoLevel[info_level]
         if lvl == InfoLevel.AUTH_LOG:
-            ret = []
-            try:
-                with open("/var/log/samba4/auth_audit.log", "r") as f:
-                    for e in f:
-                        try:
-                            entry = json.loads(e.strip())
-                        except json.JSONDecodeError:
-                            continue
-
-                        ts, extra = entry['timestamp'].split('.', 1)
-                        # add timezone info
-                        ts += extra[6:]
-                        usec = extra[:6]
-                        tv_sec = time.mktime(time.strptime(ts, "%Y-%m-%dT%H:%M:%S%z"))
-                        timestamp_tval = {
-                            "tv_sec": tv_sec,
-                            "tv_usec": int(usec)
-                        }
-                        entry['timestamp_tval'] = timestamp_tval
-                        ret.append(entry)
-
-            except FileNotFoundError:
-                self.logger.warning("SMB auth audit log does not exist "
-                                    "this is expected if users have never "
-                                    "authenticated to this server.")
-
-            return filter_list(ret, filters, options)
+            return self.middleware.call_sync('audit.query', {
+                'services': ['SMB'],
+                'query-filters': filters + [['event', '=', 'AUTHENTICATION']],
```

Comment:
Hmmm.  So you've decided to _special case_ AUTHENTICATION events and automatically add them to the filter.

Won't argue against it, but perhaps a comment with some justification would be useful.  Further, do the method docs need to be updated?

Reply:
This is actually functionally identical to previous code. With the log level change (no longer writing to this file) and a samba-level change, authentication messages are now written to the audit database. If we want output / API consistency with previous version then we need the extra filter.